tests/tests_eve/unit_tests/test_datamodels.py

```diff
@@ -0,0 +1,892 @@
+# GT4Py is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or any later
+# version. See the LICENSE.txt file at the top-level directory of this
+# distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+
+from __future__ import annotations
+
+import dataclasses
+import enum
+import inspect
+import random
+import types
+import typing
+from typing import (
+    Any,
+    Callable,
+    ClassVar,
+    Dict,
+    Generic,
+    List,
+    Literal,
+    Mapping,
+    MutableSequence,
+    Optional,
+    Sequence,
+    Set,
+    Tuple,
+    TypeVar,
+    Union,
+)
+
+import factory
+import pytest
+import pytest_factoryboy as pytfboy
+
+from eve import datamodels, utils
+
+
+T = TypeVar("T")
+
+model_factories: List[factory.Factory] = []
+invalid_model_factories: List[factory.Factory] = []
+
+
+def register_factory(
+    factory_or_name: Optional[Union[factory.Factory, str]] = None,
+    *,
+    collection: Optional[MutableSequence[factory.Factory]] = None,
+) -> Union[factory.Factory, Callable[[factory.Factory], factory.Factory]]:
+    if isinstance(factory_or_name, str):
+        name = factory_or_name
+        factory_class = None
+    else:
+        name = None
+        factory_class = factory_or_name
+
+    def _decorator(factory: factory.Factory) -> factory.Factory:
+        if collection is not None:
+            collection.append(factory)
+        return pytfboy.register(factory, name)
+
+    return _decorator(factory_class) if factory_class is not None else _decorator
+
+
+# --- Models, factories and fixtures ---
+@enum.unique
+class Kind(enum.Enum):
+    FOO = "foo"
+    BLA = "bla"
+    FIZ = "fiz"
+    FUZ = "fuz"
+
+
+@enum.unique
+class IntKind(enum.IntEnum):
+    MINUS = -1
+    ZERO = 0
+    PLUS = 1
+
+
+# NonInstantiableModel
+class NonInstantiableModel(datamodels.DataModel, instantiable=False):
+    pass
+
+
+@register_factory(collection=invalid_model_factories)
+class NonInstantiableModelFactory(factory.Factory):
+    class Meta:
+        model = NonInstantiableModel
+
+
+# EmptyModel
+class EmptyModel(datamodels.DataModel):
+    pass
+
+
+@register_factory(collection=model_factories)
+class EmptyModelFactory(factory.Factory):
+    class Meta:
+        model = EmptyModel
+
+
+# BasicFieldsModel
+class BasicFieldsModel(datamodels.DataModel):
+    bool_value: bool
+    int_value: int
+    float_value: float
+    complex_value: complex
+    str_value: str
+    bytes_value: bytes
+    kind: Kind
+    int_kind: IntKind
+    any_value: Any
+
+    class_var: ClassVar[int] = 0
+
+
+@register_factory(collection=model_factories)
+class BasicFieldsModelFactory(factory.Factory):
+    class Meta:
+        model = BasicFieldsModel
+
+    bool_value = factory.Faker("pybool")
```

Comment:
Not sure if the faked values add much value for the tests in this case.

Reply:
Not really, it was more an experiment to use it. Should I remove it?

Reply:
I think I would remove it as it only adds randomness compared to FixedBasicFieldsModelFactory

Reply:
Done